### DISCUSSION TITLE
Help with upgrading to bookworm

### DISCUSSION THREAD

**Discussion by DavidDah** [1000001101000/Debian_on_Buffalo#187] [Q&A] [answered]
Hello,
I have current configuration 
Terastation Pro II TS-HTGL/R5
root@tsprov2:~# lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 10 (buster)
Release:        10
Codename:       buster

I would like to upgrade this poor TS-HTGL/R5 to Debian bookworm, can someone suggest me apropriate steps 
how to do it.

apt update gives following errors

```shell
# apt update
Get:1 http://deb.debian.org/debian buster InRelease [122 kB]                                                                                               
Get:2 http://security.debian.org/debian-security buster/updates InRelease [34.8 kB]                                                                        
Get:3 http://http.us.debian.org/debian buster InRelease [122 kB]                                                                                           
Get:4 http://http.us.debian.org/debian buster-updates InRelease [56.6 kB]                                                                                  
Get:5 http://deb.debian.org/debian-security buster/updates InRelease [34.8 kB]                                                                             
Get:6 http://deb.debian.org/debian buster-updates InRelease [56.6 kB]                                                                                      
Get:7 http://deb.debian.org/debian buster-backports InRelease [51.4 kB]                                                                                    
Hit:8 https://deb.nodesource.com/node_12.x buster InRelease                                                                                                
Get:9 https://raw.githubusercontent.com/1000001101000/Debian_on_Buffalo/master/PPA buster InRelease [1,637 B]                                              
Get:11 http://security.debian.org/debian-security buster/updates/main Sources [410 kB]                                                                     
Get:12 http://security.debian.org/debian-security buster/updates/main Translation-en [292 kB]                                                              
Hit:10 https://packagecloud.io/ookla/speedtest-cli/debian buster InRelease                                                                                 
Get:13 http://deb.debian.org/debian-security buster/updates/main Sources [410 kB]                                                                          
Get:14 http://deb.debian.org/debian-security buster/updates/main Translation-en [292 kB]
Fetched 1,884 kB in 1min 21s (23.3 kB/s)                                                                                                                   
Reading package lists... Done
Building dependency tree       
Reading state information... Done
All packages are up to date.
N: Repository 'http://security.debian.org/debian-security buster/updates InRelease' changed its 'Suite' value from 'oldstable' to 'oldoldstable'
N: Skipping acquire of configured file 'main/binary-armel/Packages' as repository 'http://security.debian.org/debian-security buster/updates InRelease' doesn't support architecture 'armel'
N: Repository 'http://http.us.debian.org/debian buster InRelease' changed its 'Suite' value from 'oldstable' to 'oldoldstable'
N: Repository 'http://deb.debian.org/debian buster InRelease' changed its 'Suite' value from 'oldstable' to 'oldoldstable'
N: Repository 'http://deb.debian.org/debian-security buster/updates InRelease' changed its 'Suite' value from 'oldstable' to 'oldoldstable'
N: Skipping acquire of configured file 'main/binary-armel/Packages' as repository 'http://deb.debian.org/debian-security buster/updates InRelease' doesn't support architecture 'armel'
N: Skipping acquire of configured file 'contrib/binary-armel/Packages' as repository 'http://deb.debian.org/debian-security buster/updates InRelease' doesn't support architecture 'armel'
N: Skipping acquire of configured file 'non-free/binary-armel/Packages' as repository 'http://deb.debian.org/debian-security buster/updates InRelease' doesn't support architecture 'armel'
N: Repository 'http://deb.debian.org/debian buster-updates InRelease' changed its 'Suite' value from 'oldstable-updates' to 'oldoldstable-updates'
N: Repository 'http://http.us.debian.org/debian buster-updates InRelease' changed its 'Suite' value from 'oldstable-updates' to 'oldoldstable-updates'
N: Skipping acquire of configured file 'main/binary-armel/Packages' as repository 'https://deb.nodesource.com/node_12.x buster InRelease' doesn't support architecture 'armel'
```

**Comment by 1000001101000** [comment]
My best advice would be to run through the Bookworm installer. You could try following the typical dist-upgrade process but I'm not entirely sure it would succeed.

**Comment by DavidDah** [comment]
```shell
apt dist-upgrade
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
``` 
Not much luck with standard apt dist-upgrade, can You please point me to the installer for bookworm and can I run it directly from shell in existing OS or I need to go trough installation with puting drives to standard PC and do all from scratch?

**Comment by DavidDah** [comment]
Could I simply replace initrd.buffalo and  uImage.buffalo in /boot directory and reboot and then do normal installation?
Will bookworm installer owerwrite existing / partition or I can keep my config and installed packages?

**Reply by 1000001101000** [reply, marked as answer]
By dist-upgrade I mean following Debian’s guide for the process which has more steps than just calling the command. I am not confident it would succeed though.

You could load the Bookworm installer exactly how you describe. Running through the installer again you would reformat and reinstall the rootfs and /boot. You would not need to wipe out data in other partitions/arrays but you would most likely wipe out any installed packages and config files etc.


Before doing either process I’d highly recommend backing up anything you wouldn’t want to loose.

**Comment by DavidDah** [comment]
I just wanted to confirm that upgrade went flawlessly, I just coppied uImage.buffalo and initrd.buffalo to /boot directory and went trough installer from scratch, formated /boot and / partition and voila everything works.

**Comment by DavidDah** [comment]
Just reminder, If You haven't already enabled serial port by default in installer for bookworm, please add command
micro-evtd -s 000f

I have changed dynamic IP address to fixed in my bookworm and rebooted, and now it's showing error E04 on display screen,
I assume I made typo in /etc/network/interfaces file and TS can't boot anymore.
Is there some way to access to the unit and  make changes with acp_commander via MAC address since it does not request IP from DHCP
and it's not answering pings on static IP which I configured.

**Comment by 1000001101000** [comment]
Debian doesn't have buffalo's service that talks to acp_commander. It doesn't really output errors to the LCD either, I'm curious what the LCD error code is actually referring to. 

You might be able to TFTP boot into the stock firmware EM-mode then get a shell via acp_commander and try fixing things or at least enabling the serial port (which remains enabled through a soft reboot). 

I don't typically enable that serial console as part of the install since someone might actually want to use their rear serial port for a UPS or something.

**Comment by DavidDah** [comment]
Solved it with good old method of puting drives into PC and running installer.
